packages/webdriverio/tests/commands/element/__snapshots__/click.test.ts.snap

```diff
@@ -25,7 +25,7 @@ exports[`click test > should allow to left click on an element by passing a butt
     {
       "altitudeAngle": 0,
       "azimuthAngle": 0,
-      "button": 0,
+      "button": "left",
```

Comment:
Can we verify that this is accepted by the WebDriver protocol? From what I see:

> To process a pointer up or pointer down action given action item, and action:
> 
> Let button be the result of getting the property button from action item.

I can't find anywhere that it accepts `left`, `middle` or `right`.

Reply:
This works because we map it in the `action` command:
<img width="891" alt="Screenshot 2024-03-01 at 10 42 29" src="https://github.com/webdriverio/webdriverio/assets/15839059/e1bab645-0a4b-4a14-8272-825b5ab64d90">

Reply:
No need to map it twice

Reply:
Not sure what happens if you send more params though, could be that we don't map it then in which case it would be a bug in the `action` command and we simply need to fix that.

Reply:
comparing the up (top) with the the down (bottom) command, there's already a difference in what we pass. I think it should be the following: 
```js
        function mapParameters(params) {
          const buttons = {
            left: 0,
            middle: 1,
            right: 2
          }
          if (typeof params === 'string') {
            return buttons[params]
          }
          if (typeof params.button === 'string') {
            return { ...params, button: buttons[params.button] }
          }
          return params
        }
        this.sequence.push({
            type: 'pointerUp',
            ...PARAM_DEFAULTS,
            ...mapParameters(params)
        })
``` 
<img width="898" alt="Screenshot 2024-03-01 at 10 46 06" src="https://github.com/webdriverio/webdriverio/assets/15839059/2faf4f71-28ce-42e9-8875-0469c19f337c">

Reply:
I have created a PR to fix and align the mapping: #12390